src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -749,8 +749,13 @@ private protected Compilation RunGenerators(Compilation input, ParseOptions pars
             }
 
             driver ??= CreateGeneratorDriver(parseOptions, generators, analyzerConfigOptionsProvider, additionalTexts);
-            driver = driver.RunGeneratorsAndUpdateCompilation(input, out var compilationOut, out var diagnostics);
-            generatorDiagnostics.AddRange(diagnostics);
+
+            driver = driver.RunGenerators(input);
+            var results = driver.GetRunResult();
+
+            var treeRoot = this.Arguments.GeneratedFilesOutputDirectory ?? Arguments.OutputDirectory;
+            var compilationOut = input.AddSyntaxTrees(results.GeneratedTrees.Select(t => t.WithFilePath(Path.Combine(treeRoot, t.FilePath))));
```

Comment:
>AddSyntaxTrees [](http://example.com/codeflow?start=39&length=14)

Do we need to update `PostInitTrees` as well?

Reply:
No the results struct already contains them (see https://sourceroslyn.io/#Microsoft.CodeAnalysis/SourceGeneration/GeneratorDriver.cs,148)